client/src/components/wishlistIcon/WishlistIcon.js

```diff
@@ -9,15 +9,17 @@ import { selectSeriesWishlistStatus } from "../../features/wishlist/wishlist.sel
 
 const WishlistIcon = (props) => {
     const dispatch = useDispatch();
-    const { relatedUser, series, className } = props;
+    const { relatedUser, series, className, disableClick = false } = props;
 
     const wishlistFillStatus = useSelector((state) => selectSeriesWishlistStatus(state, series._id));
+    
+    const classes= `${className} ${disableClick ? " pe-none" : ""}`;
 
     const updateUserWatchlist = async (event) => {
         event.preventDefault();
 
         try {
-            const { series_ids } = await wishlistService.updateWishlist(series._id, relatedUser.email, wishlistFillStatus ? ActionType.REMOVE : ActionType.ADD);
+            const { series_ids } = await wishlistService.updateWishlist(relatedUser.email, series._id, wishlistFillStatus ? ActionType.REMOVE : ActionType.ADD);
             dispatch(updateWishlist(series_ids));
```

Comment:
It's not necessarily the current user's wishlist you need to update here. You have to send `onWishlistChange` props or something like that. I had that bug too